redis/tests/test_cache.rs

```diff
@@ -0,0 +1,307 @@
+#![cfg(all(feature = "aio", feature = "cache-aio"))]
+
+use redis::caching::CacheConfig;
+use redis::{AsyncCommands, RedisError, RedisFuture};
+use std::time::Duration;
+
+use crate::support::*;
+
+mod support;
+
+#[test]
+fn test_cache() {
```

Comment:
I'm thinking about implementing a mock multiplexed connection, would that make the statistics redundant?

Reply:
I want to test with real connection, because how client side caching works is really about how redis server responds. This way we might be able to see any issue comes up new valkey,redis versions.